python/ray/data/_internal/logical/operators/from_arrow_operator.py

```diff
@@ -0,0 +1,56 @@
+from typing import TYPE_CHECKING, List, Union
+
+import ray
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.types import ObjectRef
+
+if TYPE_CHECKING:
+    import pyarrow
+    import datasets
+
+
+class FromArrowRefs(LogicalOperator):
+    """Logical operator for `from_arrow_refs`."""
+
+    def __init__(
+        self,
+        tables: List[ObjectRef[Union["pyarrow.Table", bytes]]],
+        op_name: str = "FromArrowRefs",
+    ):
+        super().__init__(op_name, [])
+        self._tables = tables
+
+
+class FromArrow(FromArrowRefs):
+    """Logical operator for `from_arrow`."""
+
+    def __init__(
+        self, tables: List[Union["pyarrow.Table", bytes]], op_name: str = "FromArrow"
+    ):
+        super().__init__([ray.put(t) for t in tables], op_name)
```

Comment:
Should the put also be deferred here?

Reply:
Yeah we should. Since `FromArrowRefs` expects a list of `ObjectRef`s, if we defer the `put` in `FromArrow.__init__`, this creates an incompatibility in `FromArrowRefs__init__`. I can see two options for dealing with this:
(1) Make `FromArrow` no longer extend `FromArrowRefs`, and instead extend `LogicalOperator`
(2a) Modify the `tables` arg to take both `List[Union["pyarrow.Table", bytes]]` and `List[ObjectRef[Union["pyarrow.Table", bytes]]]`
(2b) Add an additional `tables_refs` arg to `FromArrowRefs.__init__`, and only use one of `tables` or `tables_ref` when initializing

Thoughts/alternatives (would apply for all other similar `put` calls in this PR)?

Reply:
Any of these options seem fine to me!

Reply:
I went with option (2a) here. This still means that we lose some of the class inheritance between some LogicalOperators (e.g. FromArrow extending FromArrowRefs), but we can still consolidate the logic to go from the logical to physical operators.